selfservice/strategy/oidc/provider_linkedin.go

```diff
@@ -0,0 +1,179 @@
+	LocalizedFirstName string `json:"localizedFirstName"`
+	ProfilePicture     struct {
+		DisplayImage struct {
+			Elements []struct {
+				Identifiers []struct {
+					Identifier string `json:"identifier"`
+				} `json:"identifiers"`
+			} `json:"elements"`
+		} `json:"displayImage~"`
+	} `json:"profilePicture"`
+	ID string `json:"id"`
+}
+
+type EmailAddress struct {
+	Elements []struct {
+		Handle struct {
+			EmailAddress string `json:"emailAddress"`
+		} `json:"handle~"`
+		HandleUrn string `json:"handle"`
+	} `json:"elements"`
+}
+
+type Introspection struct {
+	Active       bool   `json:"active"`
+	ClientID     string `json:"client_id"`
+	AuthorizedAt uint32 `json:"authorized_at"`
+	CreatedAt    uint32 `json:"created_at"`
+	ExpiresAt    uint32 `json:"expires_at"`
+	Status       string `json:"status"`
+	Scope        string `json:"scope"`
+	AuthType     string `json:"auth_type"`
+}
+
+// type APIUrl string
+
+const (
+	ProfileUrl       string = "https://api.linkedin.com/v2/me?projection=(id,localizedFirstName,localizedLastName,profilePicture(displayImage~digitalmediaAsset:playableStreams))"
+	EmailUrl         string = "https://api.linkedin.com/v2/emailAddress?q=members&projection=(elements*(handle~))"
+	IntrospectionURL string = "https://www.linkedin.com/oauth/v2/introspectToken"
+)
+
+type ProviderLinkedIn struct {
+	config *Configuration
+	reg    dependencies
+}
+
+func NewProviderLinkedIn(
+	config *Configuration,
+	reg dependencies,
+) *ProviderLinkedIn {
+	return &ProviderLinkedIn{
+		config: config,
+		reg:    reg,
+	}
+}
+
+func (l *ProviderLinkedIn) Config() *Configuration {
+	return l.config
+}
+
+func (l *ProviderLinkedIn) oauth2(ctx context.Context) *oauth2.Config {
+	return &oauth2.Config{
+		ClientID:     l.config.ClientID,
+		ClientSecret: l.config.ClientSecret,
+		Endpoint:     linkedin.Endpoint,
+		Scopes:       l.config.Scope,
+		RedirectURL:  l.config.Redir(l.reg.Config().SelfPublicURL(ctx)),
+	}
+}
+
+func (l *ProviderLinkedIn) OAuth2(ctx context.Context) (*oauth2.Config, error) {
+	return l.oauth2(ctx), nil
+}
+
+func (l *ProviderLinkedIn) AuthCodeURLOptions(r ider) []oauth2.AuthCodeOption {
+	return []oauth2.AuthCodeOption{}
+}
+
+func (l *ProviderLinkedIn) ApiCall(url string, result interface{}, exchange *oauth2.Token) error {
+	var bearer = "Bearer " + exchange.AccessToken
+	req, err := http.NewRequest(http.MethodGet, string(url), nil)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	req.Header.Add("Authorization", bearer)
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	err = json.Unmarshal(body, result)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	return nil
+}
+
+func (l *ProviderLinkedIn) Introspection(result interface{}, exchange *oauth2.Token) error {
+	resp, err := http.PostForm(string(IntrospectionURL),
```

Comment:
Here too, please use the global HTTP client :)

Reply:
Addressed, using the global http client everywhere